### DISCUSSION TITLE
how to merge two gif images horizontally and vertically

### DISCUSSION THREAD

**Discussion by kunalbali** [ImageMagick/ImageMagick#5775] [General] [answered]
Hi,

I have two gif images NH.gif and SH.gif, these gif images created via 
`convert -delay 20 -loop 0 *.jpg NH.gif ` # this is based on 2 images 
`convert -delay 20 -loop 0 *.jpg SH.gif ` # this is based on 2 images 

I want to merge these gif images side by side 
So I did

`convert  NH.gif SH.gif +append horizontal1.gif`

But, I am not getting the desired result (two gif images side by side). 
I am getting 4 images horizontally. 
![horizontal1](https://user-images.githubusercontent.com/6864691/202808190-210d7267-a7fa-422f-9c82-7808c4bd5411.gif)

Let me know how to merge two gif images side by side

Thanks

**Comment by fmw42** [comment]
See https://imagemagick.org/Usage/anim_mods/#append and scroll down to the section Layered Composition. Use the second command line in that section.

Next time post links to your GIF animations. Also identify your IM version and platform/OS.

**Comment by kunalbali** [comment]
Thanks,

I have MAC OS with
`Version: ImageMagick 7.1.0-23 Q16-HDRI x86_64 2022-01-29 https://imagemagick.org`

I did not understand the code correctly, But I tried as
`magick NH.gif -coalesce +append SH.gif -coalesce +append  horizontal2.gif `

still not getting the desired output. I know the code is wrong, could you please help me improve it? 

I also attached the two gif images (just in case you need them).

![NH](https://user-images.githubusercontent.com/6864691/202817951-c0e05758-b96a-4ae5-bcb1-13a0c6a3521b.gif)
![SH](https://user-images.githubusercontent.com/6864691/202817955-7b322a7c-1df9-428a-9925-f2ca4d9f89ac.gif)


Thanks

**Comment by snibgo** [comment]
Each input has 2 frames. They are not compressed. You want an output of 2 frames, where the first frame contains the appended first frames from both inputs, and the second frame is the appended second frames from both inputs. Is that correct?
```
magick ( NH.gif[0] SH.gif[0] +append ) ( NH.gif[1] SH.gif[1] +append ) +repage out.gif
```

**Comment by kunalbali** [comment]
Hi @snibgo

I tried your code but did not get the solution.
The output which I am looking for is attached. I just did it in the Microsoft power plant. But like to do it with ImageMagick.

![Presentation1](https://user-images.githubusercontent.com/6864691/202828498-eeb517a7-9f53-4e43-82db-e97f4c69c97b.gif)

Thanks

**Comment by fmw42** [comment, marked as answer]
What is your IM version and platform/OS?  If on Window, try

`magick ( NH.gif[0] SH.gif[0] -coalesce +append ) ( NH.gif[1] SH.gif[1] -coalesce +append ) +repage -layers optimize out.gif`

If on Linux or Mac

`magick \( NH.gif[0] SH.gif[0] -coalesce +append \) \( NH.gif[1] SH.gif[1] -coalesce +append \) +repage -layers optimize out.gif`

![out](https://user-images.githubusercontent.com/13458218/202829354-e41b281b-8050-4a95-9537-28dd4b2bc8cd.gif)

**Reply by kunalbali** [reply]
Thanks, it worked.
by the way 

I am using Mac OS with
Version: ImageMagick 7.1.0-23 Q16-HDRI x86_64 2022-01-29 https://imagemagick.org

I used 
`magick \( NH.gif'[0]' SH.gif'[0]' -coalesce +append \) \( NH.gif'[1]' SH.gif'[1]' -coalesce +append \) +repage -layers optimize out.gif`
